### DISCUSSION TITLE
Plugin does not receive a PlayerPluginMessage

### DISCUSSION THREAD

**Discussion by LignumTseshka** [Minestom/Minestom#1764] [Help] [answered]
I've been adapting a mod's plugin for Minestom and trying to receive a plugin message which I'm sure exitsts.

```
MinecraftServer.getGlobalEventHandler().addListener(EventListener.builder(PlayerPluginMessageEvent.class) 
                .handler((event) -> {
                    getLogger().info(event.getIdentifier());
                    getLogger().info(event.getMessageString());
                })
                .build());
```

But whenever I'm trying to see all the coming packets or the plugin message packets only, all I can find are these two 
```
[INFO] [15:20:10] [minestom] | minecraft:register
[INFO] [15:20:10] [minestom] | fml:loginwrapperforge:tier_sortingfml:handshakeplasmo:voicefml:playforge:split
[INFO] [15:20:10] [minestom] | minecraft:brand
[INFO] [15:20:10] [minestom] | forge
```

**Comment by LooFifteen** [comment, marked as answer]
Which mod is it? It might require a plugin message from the server first.

**Reply by LignumTseshka** [reply]
It's called Plasmo Voice
https://github.com/plasmoapp/plasmo-voice/

**Reply by LooFifteen** [reply]
The server listens for the channel registry packet, which is being sent as per your logging:
```
[INFO] [15:20:10] [minestom] | minecraft:register
[INFO] [15:20:10] [minestom] | fml:loginwrapperforge:tier_sortingfml:handshakeplasmo:voicefml:playforge:split
                                                                              ┖──────────┴─ Here
```
It [listens for the registry of the `plasmo:voice` channel](https://github.com/plasmoapp/plasmo-voice/blob/9b75d8576f6c67c0b540c15e684162609f0d1732/server/paper/src/main/java/su/plo/voice/paper/connection/PaperServerChannelHandler.java#L55-L7), causing the [channel registry handler](https://github.com/plasmoapp/plasmo-voice/blob/9b75d8576f6c67c0b540c15e684162609f0d1732/server/common/src/main/java/su/plo/voice/server/connection/BaseServerChannelHandler.java#L38) to send a [player info request packet](https://github.com/plasmoapp/plasmo-voice/blob/9b75d8576f6c67c0b540c15e684162609f0d1732/protocol/src/main/java/su/plo/voice/proto/packets/tcp/clientbound/PlayerInfoRequestPacket.java#L17) back to the player.

Hope this helps!
